src/main/java/Tree/BinarySearchTree.java

```diff
@@ -6,7 +6,7 @@
 
 
     public static void add(Node root, int value) {
-        if (value <= root.getValue()) {
+        if (value <= (int)root.getValue()) {

```

Comment:
The fact that you need to cast these indicates you're not using generics correctly.